07-routing-gossip.md

```diff
@@ -0,0 +1,252 @@
+insufficent length to contain `alias`.
+
+The receiving node SHOULD fail the connection if `signature` is
+invalid or incorrect for the entire message including unknown fields
+following `alias`, and MUST NOT further process the message.  The
+receiving node SHOULD ignore `ipv6` if `port` is zero.  It MUST ignore
+the contents of `pad`.
+
+The receiving node SHOULD ignore the message if `node-id` is not
+previously known from a `channel_announcement` message, or if
+`timestamp` is not greater than than the last-received
+`node_announcement` from this `node-id`.  Otherwise, if the
+`timestamp` is greater than the last-received `node_announcement` from
+this `node-id` the receiving node SHOULD queue the message for
+rebroadcasting.
+
+The receiving node MAY use `rgb` and `alias` to reference nodes in interfaces, but SHOULD insinuate their self-signed origin.
+
+### Rationale
+
+RFC 4291 section 2.5.5.2 described IPv4 addresses like so:
+
+```
+|                80 bits               | 16 |      32 bits        |
++--------------------------------------+--------------------------+
+|0000..............................0000|FFFF|    IPv4 address     |
++--------------------------------------+----+---------------------+
+```
+
+## The `channel_update` message
+
+After a channel has been initially announced, each side independently
+announces its fees and minimum expiry for HTLCs.  It uses the 8-byte
+channel shortid which matches the `channel_announcement` and one byte
+to indicate which end this is.  It can do this multiple times, if
+it wants to change fees.
+
+1. type: 258 (`MSG_CHANNEL_UPDATE`)
+2. data:
+	* [64:signature]
+	* [4:blockheight]
+	* [3:blockindex]
+	* [1:outputindex]
+	* [4:timestamp]
+	* [1:side]
+	* [1:pad]
+    * [2:expiry]
+	* [4:htlc-minimum-msat]
+	* [4:fee-base-msat]
+	* [4:fee-proportional-millionths]
+
+### Requirements
+
+The creating node MUST set `signature` to the signature of the
+double-SHA of the entire remaining packet after `signature` using its own `node-id`.
+
+The creating node MUST set `blockheight`, `blockindex` and `outputindex` to
+match those in the already-sent `channel_announcement` message, and MUST set `side` to 0 if the creating node is `node-id-1` in that message, otherwise 1.
+
+The creating node MUST set `timestamp` to greater than zero, and MUST set it to greater than any previously-sent `channel_update` for this channel, and MUST set `pad` to zero.
+
+It MUST set `expiry` to the number of blocks it will subtract from an incoming HTLC's `expiry`.  It MUST set `htlc-minimum-msat` to the minimum HTLC value it will accept, in millisatoshi.  It MUST set `fee-base-msat` to the base fee it will charge for any HTLC, in millisatoshi, and `fee-proportional-millionths` to the amount it will charge per millionth of a satoshi.
+
+The node SHOULD accept HTLCs which pay a fee equal or greater than:
+
+	fee-base-msat + htlc-amount-msat * fee-proportional-millionths / 1000000
```

Comment:
Right, but that is the node's local policy isn't it? Either he has a grace period, or he is rejecting things that'd make him some fees. So I'm not sure we need to specify this here.